src/Ledger/Utxow.lagda

```diff
@@ -83,11 +91,18 @@ data _⊢_⇀⦇_,UTXOW⦈_ where
         open UTxOState s
         witsKeyHashes     = mapˢ hash (dom vkSigs)
         witsScriptHashes  = mapˢ hash scripts
+        inputHashes       = getInputHashes tx utxo
+        refScriptHashes   = mapˢ hash (refScripts tx utxo)
+        neededHashes      = scriptsNeeded utxo txb
+        txdatsHashes      = dom (txdats .proj₁)
+        allOutHashes      = getDataHashes (range txouts)
     in
     ∙  ∀[ (vk , σ) ∈ vkSigs ] isSigned vk (txidBytes txid) σ
-    ∙  ∀[ s ∈ scriptsP1 ] validP1Script witsKeyHashes txvldt s
+    ∙  ∀[ s ∈ mapPartial isInj₁ (txscripts tx utxo) ∩ scriptsP1 ] validP1Script witsKeyHashes txvldt s
     ∙  witsVKeyNeeded utxo txb ⊆ witsKeyHashes
-    ∙  scriptsNeeded utxo txb ≡ᵉ witsScriptHashes
+    ∙  (neededHashes ＼ refScriptHashes) ≡ᵉ witsScriptHashes
+    ∙  inputHashes ⊆ txdatsHashes
+    ∙  txdatsHashes ⊆ inputHashes ∪ getDataHashes (range (utxo ∣ refInputs))
```

Comment:
This also needs to include the hashes in `txouts` in the RHS

Reply:
Okay, I'll recompile the pdf in case there's anything new and then try to reconcile this comment with what's in Fig 6.  If/when I can't decide exactly what should go here in the formalization, I'll ping you on Slack.

Reply:
@WhatisRT It's not exactly clear what you mean by "This" and "RHS," but if I had to guess, you are saying that the line `(neededHashes ＼ refScriptHashes) ≡ᵉ witsScriptHashes` should be changed to `(neededHashes ＼ refScriptHashes) ≡ᵉ witsScriptHashes ∪ txdatsHashes`.  Is that correct?

Reply:
No, by this I mean the line where I made the comment, i.e. `txdatsHashes ⊆ inputHashes ∪ getDataHashes (range (utxo ∣ refInputs))`, the RHS being `inputHashes ∪ getDataHashes (range (utxo ∣ refInputs))`. If you compare this to the Babbage spec, it's effectively a union of three things, `inputHashes`, the hashes of `allOuts` and those of `refInputs`. You're missing those that belong to `allOuts` here.

Reply:
Thanks for the clarification.